packages/blade/src/components/Radio/__tests__/RadioGroup.native.test.tsx

```diff
@@ -0,0 +1,208 @@
+    expect(queryByText(labelText)).toBeTruthy();
+    expect(queryByText(helpText)).toBeTruthy();
+  });
+
+  it('should propagate isDisabled prop to child radios', () => {
+    const labelText = 'Select fruits';
+    const { getAllByA11yRole } = renderWithTheme(
+      <RadioGroup isDisabled label={labelText}>
+        <Radio value="apple">Apple</Radio>
+        <Radio value="mango">Mango</Radio>
+        <Radio value="orange">Orange</Radio>
+      </RadioGroup>,
+    );
+
+    const radios = getAllByA11yRole('radio');
+    radios.forEach((radio) => {
+      expect(radio.props.accessibilityState.disabled).toBeTruthy();
+    });
+  });
+
+  it('should propagate name prop to child radios', () => {
+    const labelText = 'Select fruits';
+    const { getAllByA11yRole } = renderWithTheme(
+      <RadioGroup isDisabled label={labelText} name="fruits">
+        <Radio value="apple">Apple</Radio>
+        <Radio value="mango">Mango</Radio>
+        <Radio value="orange">Orange</Radio>
+      </RadioGroup>,
+    );
+
+    const radios = getAllByA11yRole('radio');
+    radios.forEach((radio) => {
+      expect(radio.props.name).toBe('fruits');
+    });
+  });
+
+  it('should render helpText of individual radios when inside group', () => {
+    const labelText = 'Select fruits';
+    const helpText = 'Select one';
+
+    const { queryByText } = renderWithTheme(
+      <RadioGroup helpText={helpText} label={labelText}>
+        <Radio helpText="Apple help" value="apple">
+          Apple
+        </Radio>
+        <Radio helpText="Mango help" value="mango">
+          Mango
+        </Radio>
+      </RadioGroup>,
+    );
+
+    expect(queryByText(helpText)).toBeTruthy();
+    expect(queryByText('Apple help')).toBeTruthy();
+    expect(queryByText('Mango help')).toBeTruthy();
+  });
+
+  it('should render errorText when validationState is set to error', () => {
+    const labelText = 'Select fruits';
+    const helpText = 'Select one';
+    const errorText = 'Invalid selection';
+
+    const { getAllByA11yRole, queryByText } = renderWithTheme(
+      <RadioGroup
+        helpText={helpText}
+        errorText={errorText}
+        label={labelText}
+        validationState="error"
+      >
+        <Radio value="apple">Apple</Radio>
+        <Radio value="mango">Mango</Radio>
+        <Radio value="orange">Orange</Radio>
+      </RadioGroup>,
+    );
+
+    expect(queryByText(helpText)).toBeFalsy();
+    expect(queryByText(errorText)).toBeTruthy();
+
+    const radios = getAllByA11yRole('radio');
+    radios.forEach((radio) => {
+      expect(radio.props.accessibilityInvalid).toBeTruthy();
+    });
+  });
+
+  it('should work in uncontrolled mode', () => {
+    const labelText = 'Select fruits';
+    const fn = jest.fn();
+    const { getByA11yState, getAllByA11yRole } = renderWithTheme(
+      <RadioGroup
+        name="fruits"
+        label={labelText}
+        defaultValue="apple"
+        onChange={({ value }) => fn(value)}
```

Comment:
nit: directly pass `onChange` mock function

```suggestion
        onChange={onChange}
```